server/node.go

```diff
@@ -235,8 +235,11 @@ func NewNode(ctx storage.StoreContext, recorder *status.MetricsRecorder, stopper
 }
 
 // context returns a context encapsulating the NodeID.
-func (n *Node) context() context.Context {
-	return log.Add(context.Background(), log.NodeID, n.Descriptor.NodeID)
+func (n *Node) context(ctx context.Context) context.Context {
+	if ctx == nil {
+		ctx = context.Background()
```

Comment:
I feel pretty strongly about pushing the "ugliness" of conjuring up a context to the caller, so I'd rather we apply that treatment to storage.Store as well.